x-pack/plugins/apm/server/lib/settings/apm_indices/get_apm_indices.ts

```diff
@@ -64,19 +68,38 @@ export async function getApmIndices({
       savedObjectsClient
     );
     const apmIndicesConfig = getApmIndicesConfig(config);
-    return merge({}, apmIndicesConfig, apmIndicesSavedObject);
+    return mergeApmIndicesConfigs(apmIndicesConfig, apmIndicesSavedObject);
   } catch (error) {
     return getApmIndicesConfig(config);
   }
 }
 
-const APM_UI_INDICES: ApmIndicesName[] = [
-  'apm_oss.sourcemapIndices',
-  'apm_oss.errorIndices',
-  'apm_oss.onboardingIndices',
-  'apm_oss.spanIndices',
-  'apm_oss.transactionIndices',
-  'apm_oss.metricsIndices',
+function mergeApmIndicesConfigs(
+  apmIndicesConfig: ApmIndicesConfig,
+  apmIndicesSavedObject: PromiseReturnType<typeof getApmIndicesSavedObject>
+) {
+  return APM_UI_INDICES.reduce(
+    (mergedConfigs, savedIndexConfig) => {
+      const savedApmIndex =
+        apmIndicesSavedObject[savedIndexConfig[0]] ||
+        apmIndicesSavedObject[savedIndexConfig[1]]; // TODO remove deprecated apm_oss support in 8.0
+      return {
+        ...mergedConfigs,
+        [savedIndexConfig[0]]: savedApmIndex,
+        [savedIndexConfig[1]]: savedApmIndex,
+      };
+    },
+    { ...apmIndicesConfig }
+  );
+}
+
+const APM_UI_INDICES: Array<[ApmIndicesName, ApmIndicesName]> = [
+  ['xpack.apm.sourcemapIndices', 'apm_oss.sourcemapIndices'],
```

Comment:
Do we still need `apm_oss.*` settings? I thought by using `renameFromRoot` we could refer to them with their new names (`xpack.apm.*`)

Reply:
`renameFromRoot` will let you reference them from the plugin context, but APM UI has a few places where configs are copied and referenced for apm-internal use by the APM event client and in the setup object, so we need to explicitly support both configs there until 8.0. In this case, it's just a list/mapping of new and deprecated index configs for use in the index settings functionality, in case deprecated names exist in the saved object.

Reply:
removed these after migrating the saved objects mappings